### ISSUE TITLE
Evaluation Results for LLaVA1.5 on COCO Dataset

### ISSUE THREAD

**Issue by Mongodiem** [GasolSun36/MVP#3]
Hi,

I've completed the evaluation for your project using LLaVA1.5, and here are my results:

**Metrics:**

| Model                                | LLaVA1.5 coco_pope_popular | LLaVA1.5 coco_pope_random | LLaVA1.5 coco_pope_adversarial |
|--------------------------------------|----------------------------|----------------------------|---------------------------------|
| Accuracy                             | 0.8554                     | 0.8991                     | 0.8069                          |
| Precision                            | 0.8622                     | 0.9442                     | 0.7843                          |
| Recall                               | 0.8460 (-11%)              | 0.8483 (-9%)               | 0.8465 (-10%)                   |
| F1 Score                             | 0.8540                     | 0.8937                     | 0.8142                          |

**Code Used for Evaluation:**

```python
import os
import json
from tqdm import tqdm

def load_json_lines(file_path):
    try:
        with open(os.path.expanduser(file_path), 'r') as file:
            return [json.loads(line) for line in file]
    except Exception as e:
        print(f"Failed to load or parse {file_path}: {e}")
        return []

def compute_probabilities(weights, *args):
    # 计算 Yes 概率
    yes_prob = sum(weight * data['aggretion_logits']['Yes'] for weight, data in zip(weights, args))
    # 计算 No 概率
    no_prob = sum(weight * data['aggretion_logits']['No'] for weight, data in zip(weights, args))
    print(f"yes_prob: {yes_prob}, no_prob: {no_prob}")  # 打印每个视角的概率
    return yes_prob, no_prob

def evaluate_predictions(weights, gt_data, predictions):
    true_pos, true_neg, false_pos, false_neg, unknown, yes_answers = 0, 0, 0, 0, 0, 0
    
    for gt, bottom_up, top_down, normal in zip(gt_data, *predictions):
        gt_answer = gt['label'].lower().strip()
        yes_prob, no_prob = compute_probabilities(weights, bottom_up, top_down, normal)
        gen_answer = 'yes' if yes_prob > no_prob else 'no'
        
        if gt_answer in {'yes', 'no'}:
            if gen_answer == gt_answer:
                true_pos += 1 if gen_answer == 'yes' else true_neg
            else:
                false_neg += 1 if gt_answer == 'yes' else false_pos
            if gen_answer == 'yes':
                yes_answers += 1
        else:
            unknown += 1
    # 只在最终输出统计结果
    print(f"Final counts -> true_pos: {true_pos}, true_neg: {true_neg}, false_pos: {false_pos}, false_neg: {false_neg}, unknown: {unknown}, yes_answers: {yes_answers}")
    
    total_questions = len(gt_data)
    precision = true_pos / (true_pos + false_pos) if true_pos + false_pos > 0 else 0
    recall = true_pos / (true_pos + false_neg) if true_pos + false_neg > 0 else 0
    f1 = 2 * precision * recall / (precision + recall) if precision + recall > 0 else 0
    accuracy = (true_pos + true_neg) / total_questions if total_questions > 0 else 0
    yes_proportion = yes_answers / total_questions if total_questions > 0 else 0
    unknown_prop = unknown / total_questions if total_questions > 0 else 0
    
    return accuracy, precision, recall, f1, yes_proportion, unknown_prop

def main():
    # 直接设置文件路径和参数
    gt_files = "/home/MLLMs/MVP/benchmark/POPE/coco/coco_pope_popular.json"
    gen_files_bottom_up = "/home/MLLMs/MVP/output/coco_pope_popular_caption_llava_bottom-up_test2.jsonl"
    gen_files_top_down = "/home/MLLMs/MVP/output/coco_pope_popular_caption_llava_top-down_test2.jsonl"
    gen_files_normal = "/home/MLLMs/MVP/output/coco_pope_popular_caption_llava_regular_test2.jsonl"
    
    a = 0.4
    b = 0.4
    c = 0.2
    weights = (a, b, c)

    # 加载数据
    gt_data = load_json_lines(gt_files)
    gen_files_bottom_up_data = load_json_lines(gen_files_bottom_up)
    gen_files_top_down_data = load_json_lines(gen_files_top_down)
    gen_files_normal_data = load_json_lines(gen_files_normal)
    
    # 评估
    results = evaluate_predictions(weights, gt_data, (gen_files_bottom_up_data, gen_files_top_down_data, gen_files_normal_data))
    
    # 打印结果
    print(f"Accuracy: {results[0]*100:.2f}%")
    print(f"Precision: {results[1]*100:.2f}%")
    print(f"Recall: {results[2]*100:.2f}%")
    print(f"F1 Score: {results[3]*100:.2f}%")
    print(f"Yes Proportion: {results[4]*100:.2f}%")
    print(f"Unknown Proportion: {results[5]*100:.2f}%")

if __name__ == "__main__":
    main()
